### ISSUE TITLE
Support for easy SIGINT handler

### ISSUE THREAD

**Issue by WebFreak001** [Dechcaudron/dli#4]
so Ctrl-C doesn't close the application, but usually they are weird to write using the C apis

**Comment by Dechcaudron** [comment]
just wondering, why shouldn't Ctrl-C close the application?

**Comment by WebFreak001** [comment]
if you want to do cleanup, stopping servers, when autocompleting simply remove what was typed, etc

**Comment by Dechcaudron** [comment]
ok, so Ctrl-C should be able to gracefully stop the app. Is that what you mean?

**Comment by WebFreak001** [comment]
I want something like
```d
handleInterrupt(() {
  writeln("no");
});
```

**Comment by Dechcaudron** [comment]
I'll see how I go about it. I've never worked with OS signals before and don't know the apis. Still, handling the Ctrl-C signal should not digest it. The application should exit gracefully, but exit regardless.

**Comment by WebFreak001** [comment]
see nodejs shell, it doesn't exit on first ctrl-c either, only clears the input first and exits on the second one

**Comment by Dechcaudron** [comment]
yeah, I've come across a couple apps that do that as well. This is definitely useful, but I need to come up with a good API. Goes to my TODO.

**Comment by Dechcaudron** [comment]
Okay, so I've been doing some research about this. Off the top of my head, I'd say the way to go is to create a separate library which takes care of handling IPC signals in POSIX systems, since not only apps that use cli may benefit from custom signal handling. The approach is exactly the same to handle SIGINT or virtually any other signal, so that'd make it more flexible as well. What do you think?
